packages/react-components/src/components/TextFieldWithMention/mentionTagUtils.ts

```diff
@@ -376,29 +385,30 @@ export const updateHTML = (props: UpdateHTMLProps): { updatedHTML: string; updat
             if (startIndex !== tag.plainTextBeginIndex && startIndex !== closingTagInfo.plainTextEndIndex) {
               // mention tag should be deleted when user tries to edit it in the middle
               result += htmlText.substring(lastProcessedHTMLIndex, tag.openTagIndex) + processedChange;
-              changeNewEndIndex = tag.plainTextBeginIndex + processedChange.length;
+              changeNewEndIndex = tag.plainTextBeginIndex + unEscapeHtmlCharacters(processedChange).length;
               lastProcessedHTMLIndex = closingTagInfo.closeTagIdx + closingTagInfo.closeTagLength;
             } else if (startIndex === tag.plainTextBeginIndex) {
               // non empty change at the beginning of the mention tag to be added before the mention tag
               result += htmlText.substring(lastProcessedHTMLIndex, tag.openTagIndex) + processedChange;
-              changeNewEndIndex = tag.plainTextBeginIndex + processedChange.length;
+              changeNewEndIndex = tag.plainTextBeginIndex + unEscapeHtmlCharacters(processedChange).length;
               lastProcessedHTMLIndex = tag.openTagIndex;
             } else if (startIndex === closingTagInfo.plainTextEndIndex) {
               // non empty change at the end of the mention tag to be added after the mention tag
               result +=
                 htmlText.substring(lastProcessedHTMLIndex, closingTagInfo.closeTagIdx + closingTagInfo.closeTagLength) +
                 processedChange;
-              changeNewEndIndex = closingTagInfo.plainTextEndIndex + processedChange.length;
+              changeNewEndIndex = closingTagInfo.plainTextEndIndex + unEscapeHtmlCharacters(processedChange).length;
               lastProcessedHTMLIndex = closingTagInfo.closeTagIdx + closingTagInfo.closeTagLength;
             }
             processedChange = '';
           } else {
+            // When change is empty, it means that this change is a deletion
             const updateMentionTagResult = handleMentionTagUpdate({
               htmlText,
               oldPlainText,
               lastProcessedHTMLIndex,
               processedChange,
-              change,
+              change: change,
```

Comment:
minor: parameter name is not required here as parameter and value names are the same